mcp/content.go

```diff
@@ -25,12 +25,19 @@ type TextContent struct {
 }
 
 func (c *TextContent) MarshalJSON() ([]byte, error) {
```

Comment:
Thank you for fixing this oversight.
The same fix should apply to other content types, but you do not need to do that in this CL--we appreciate the fix as it is. However, could you please add

```go
// TODO(findleyr): update JSON marshalling of all content types to preserve required fields.
// (See [TextContent.MarshalJSON], which handles this for text content).
```

near the top of content.go to address this bug for other content types?